sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -881,7 +862,8 @@ object StateStore extends Logging {
             case NonFatal(e) =>
               logWarning(log"Error managing ${MDC(LogKeys.STATE_STORE_PROVIDER, provider)}, " +
                 log"stopping management thread", e)
-              threadPoolException.set(e)
```

Comment:
We should probably note that the side effect of this is that an exception being triggered will cause an unload on the partition only when that partition detects the error and not before.

So its a tradeoff:
- if the exception is specific to the partition, we don't penalize other partitions (potentially from separate queries)
- if the exception does apply in general, then each partition would need to see it for the unload to be triggered

Reply:
Yes, that's true. I think that trade-off is fine. If there really is some general, global error (I have never seen this; can you give an example?), then presumably every thread will see it and will be unloaded.

Reply:
yea - the case of the error applying to partition or query has higher chance than for all partitions on a given node

Reply:
Comment added.